def contrived(a, &f)
# the block can be accessed through f
f.call(a)
# but yield also works !
yield(a)
end
# this works
contrived(25) {|x| puts x}
# this raises ArgumentError, because &f
# isn‚t really an argument - it‚s only there
# to convert a block
contrived(25, lambda {|x| puts x})
